ide/api.java.classpath/src/org/netbeans/api/java/classpath/ClassPath.java

```diff
@@ -1415,7 +1415,9 @@ public void fileDataCreated(FileEvent fe) {
 
         @Override
         public void fileChanged(FileEvent fe) {
-            processEvent(fe);
+            if (!fe.getFile().isValid()) {
```

Comment:
This is in fact wrong and incompatible change.
The `ClassPath.PROP_ROOTS` must be fired on content change.